vmdb/spec/requests/api/tags_spec.rb

```diff
@@ -40,331 +42,313 @@ def app
-
-      @success = run_post(@vm1_tags_url, gen_request(:assign, :name => @tag1[:path]))
-
-      expect(@success).to be_true
-      expect(@code).to eq(200)
-      expect(@result).to have_key("results")
-      results = @result["results"]
-      expect(results.size).to eq(1)
-      result = results.first
-      expect(result["success"]).to be_true
-      expect(result["href"]).to match(@vm1_url)
-      expect(result["tag_category"]).to eq(@tag1[:category])
-      expect(result["tag_name"]).to eq(@tag1[:name])
+    context "assigns a tag to a Vm by name path" do
+      let(:tag_results) do
+        [{:success => true, :href => vm1_url, :tag_category => tag1[:category], :tag_name => tag1[:name]}]
+      end
+
+      before do
+        api_basic_authorize subcollection_action_identifier(:vms, :tags, :assign)
+
+        run_post(vm1_tags_url, gen_request(:assign, :name => tag1[:path]))
+      end
+
+      it "tagging_result" do
+        expect_tagging_result(:tag_results)
+      end
     end
 
-    it "assigns a tag to a Vm by href" do
-      update_user_role(@role, subcollection_action_identifier(:vms, :tags, :assign))
-      basic_authorize @cfme[:user], @cfme[:password]
-
-      tag = Tag.find_by_name(@tag1[:path])
-      @success = run_post(@vm1_tags_url, gen_request(:assign, :href => "#{@cfme[:tags_url]}/#{tag.id}"))
-
-      expect(@success).to be_true
-      expect(@code).to eq(200)
-      expect(@result).to have_key("results")
-      results = @result["results"]
-      expect(results.size).to eq(1)
-      result = results.first
-      expect(result["success"]).to be_true
-      expect(result["href"]).to match(@vm1_url)
-      expect(result["tag_category"]).to eq(@tag1[:category])
-      expect(result["tag_name"]).to eq(@tag1[:name])
+    context "assigns a tag to a Vm by href" do
+      let(:tag_results) do
+        [{:success => true, :href => vm1_url, :tag_category => tag1[:category], :tag_name => tag1[:name]}]
+      end
+
+      before do
+        api_basic_authorize subcollection_action_identifier(:vms, :tags, :assign)
+
+        tag = Tag.find_by_name(tag1[:path])
+        run_post(vm1_tags_url, gen_request(:assign, :href => tags_url(tag.id)))
+      end
+
+      it "tagging_result" do
+        expect_tagging_result(:tag_results)
+      end
     end
 
-    it "assigns an invalid tag by href to a Vm" do
-      update_user_role(@role, subcollection_action_identifier(:vms, :tags, :assign))
-      basic_authorize @cfme[:user], @cfme[:password]
+    context "assigns an invalid tag by href to a Vm" do
+      before do
+        api_basic_authorize subcollection_action_identifier(:vms, :tags, :assign)
 
-      @success = run_post(@vm1_tags_url, gen_request(:assign, :href => "#{@cfme[:tags_url]}/999999"))
+        run_post(vm1_tags_url, gen_request(:assign, :href => invalid_tag_url))
+      end
 
-      expect(@success).to be_false
-      expect(@code).to eq(404)
+      it "resource_not_found" do
+        expect_resource_not_found
```

Comment:
Postulating here (not asking for a change).
Wonder if you need all these `expect_resource_...` methods. Probably a number of the single liners checking status could be merged and parameterized (aka `expect_resource :not_found`)